### PR TITLE
Explain what a mask even is

### PR DESCRIPTION
**Pull request by Lypheo**
Moved the dehalo example down because it didn’t fit well at the top.

### CODE REVIEW

File: encoding/masking-limiting-etc.md [hunk 1]
```diff
@@ -22,9 +22,28 @@ This article will cover:
...
+e.g. with an edgemask,
+and merging the filtered clip with the unfiltered clip
+according to the mask,
+such that the masked areas are taken from the unfiltered clip,
+and the unmasked areas are taken from the filtered clip.
+In effect,
+this applies the filtering only to the unmasked areas of the clip,
+leaving the masked details/edges intact.
+(In this context,
+“masked” means that the pixels in question are white in the mask clip.)
```
Comment (FichteFoll) [inline encoding/masking-limiting-etc.md]: This parenthesized explanation doesn't really help with understanding the internals because it's too unspecific. Either be more specific on how it works or leave it out entirely since the more detailed explanation is in line 55.

File: encoding/masking-limiting-etc.md [hunk 2]
```diff
@@ -33,8 +52,47 @@ and one of these formats:
 `GRAY16` (16 bits integer),
 or `GRAYS` (single precision floating point).
 
-Vapoursynth includes some basic tools for manipulating masks as well:
+#### [std.MaskedMerge][]
```
Comment (FichteFoll) [inline encoding/masking-limiting-etc.md:53]: Eh, going from from h1 to h4 is suboptimal, but using h2 for a filter seems superfluous.

File: encoding/masking-limiting-etc.md [hunk 3]
```diff
@@ -33,8 +52,47 @@ and one of these formats:
...
+like this
+(the variables refer to the value of the same pixel in each clip):
+
+```py
+output = clipa * (MAX_VALUE - mask) + clipb * mask # in 8 bpp for example, MAX_VALUE would be 255
+```
+
+In simpler terms:
+for brighter areas in the mask,
+the output will come from clipb,
```
Comment (FichteFoll) [inline encoding/masking-limiting-etc.md]: I suggest using inline code blocks when referring to variable names, so `` `clipb` ``. (Actually I went ahead and did that.)

File: encoding/masking-limiting-etc.md [hunk 4]
```diff
@@ -138,6 +196,34 @@ see [kageru's blog post][].
 
 [kageru's blog post]: https://kageru.moe/blog/article/edgemasks
 
+I would also lump the range mask
```
Comment (FichteFoll) [inline encoding/masking-limiting-etc.md]: With how many different authors we have now, using `I` seems odd.
That's just for futore thought, however, since you didn't change that in this PR.

Comment (Lypheo) [reply encoding/masking-limiting-etc.md]: Perhaps put the author's name (Zastin, I think) in parantheses after the “I”?

File: encoding/masking-limiting-etc.md (hunk 2 above)
Comment (Lypheo) [reply encoding/masking-limiting-etc.md:53]: So what do you suggest, then? 🤔

Comment (FichteFoll) [reply encoding/masking-limiting-etc.md:53]: If I had a suggestion, I would have already given it. 😄 

I say leave it as h4 for now, because it's like that for the rest of the file already and looks good enough. Until we figure out something better, this will most certainly do. h3 is still too prominent in the default style. You can resolve this.

File: encoding/masking-limiting-etc.md (hunk 4 above)
Comment (FichteFoll) [reply encoding/masking-limiting-etc.md]: Either that, or rephrase to be less opinionated and more generally applicable, I say. Or, well, just remove the personal apsect of the message and do a well-informed best practice suggestion.

**Comment by FichteFoll** [comment]
I suggest explaining std.Merge before std.MaskedMerge because the latter is the more complicated form of the former, thus allowing to explain the general concept for the Merge function and then explaining how a mask clip helps.

**Comment by Lypheo** [comment]
> I suggest explaining std.Merge before std.MaskedMerge because the latter is the more complicated form of the former, thus allowing to explain the general concept for the Merge function and then explaining how a mask clip helps.

I’m not sure if that order would make more sense, because I deliberately placed MaskedMerge right after the general description of masking in order to keep the explanation cohesive. Putting Merge above it doesn’t help the flow much IMO.

**Comment by FichteFoll** [comment]
Hm, but what about explaining std.Merge before masking concepts? I mean you need the concept of merging two clips before you can merge them selectively.

File: encoding/masking-limiting-etc.md (hunk 4 above)
Comment (OrangeChannel) [reply encoding/masking-limiting-etc.md]: I changed around the wording on all the uses of `I`, let me know what you guys think.

**Comment by OrangeChannel** [comment]
> Hm, but what about explaining std.Merge before masking concepts? I mean you need the concept of merging two clips before you can merge them selectively.

The way it's organized right now, this would require moving the `Single and multi-clip adjustments with std.Expr and friends` section above the `Masking` one.

**Comment by FichteFoll** [comment]
Well, we can revisit this later if someone sees the need. For now, let's get this shipped. 🚀
